backend/server/api/services/todos.service.js

```diff
@@ -18,6 +18,39 @@ export class ToDoService {
     const newTodo = allTodos.child(newTodoRef.key).toJSON();
     return { ...newTodo, key: newTodoRef.key };
   }
+
+  async search(searchTodo) {
+    let matchTodos = [];
+    await this.db.ref("todo").once(
+      "value",
+      function (snapshot) {
+        const todos = snapshot.val();
+        for (const property in todos) {
```

Comment:
Have you researched Firebase's querying functionality? https://firebase.google.com/docs/database/admin/retrieve-data#section-queries

Reply:
I didn't. Interesting!